### PR TITLE
Fill closed arrows in element_line()

### LINKED ISSUE #2922: Closed arrow on `axis.line` is not filled
The arrow on `axis.line` is not filled when the type is set to "closed", however other arrows of the same type are filled. See example.

``` r
library(tidyverse)

arrow = arrow(angle=10, type = "closed")
ggplot(iris, aes(x=Sepal.Length, y=Sepal.Width, color=Species)) +
    geom_point() +
    theme(axis.line = element_line(arrow=arrow)) +
    annotate("segment", x = 3, xend = min(iris$Sepal.Length), y = 4, yend = 3, arrow=arrow)
```

<sup>Created on 2018-10-01 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### PR DESCRIPTION
**Pull request by yutannihilation**
Fixes #2922 

Currently, `element_grob.element_line()` doesn't specify `fill` in `gpar()`. This PR specifies it to fill arrows.

``` r
library(ggplot2)

arrow = arrow(angle=10, type = "closed")
ggplot(iris, aes(x=Sepal.Length, y=Sepal.Width, color=Species)) +
  geom_point() +
  theme(axis.line = element_line(arrow=arrow)) +
  annotate("segment", x = 3, xend = min(iris$Sepal.Length), y = 4, yend = 3, arrow=arrow)
```

<sup>Created on 2018-10-02 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Note that, for consistency with `geom_segment()` and `geom_curve()`, `element_line()` might needs `arrow.fill` parameter, but it seems a bit harder to introduce a new parameter to theme. So, users need to rely on [this workaround](https://github.com/tidyverse/ggplot2/issues/2922#issuecomment-425943633) if they want different colours.

### CODE REVIEW

File: R/theme-elements.r [hunk 1]
```diff
@@ -226,9 +226,11 @@ element_grob.element_line <- function(element, x = 0:1, y = 0:1,
   default.units = "npc", id.lengths = NULL, ...) {
 
   # The gp settings can override element_gp
-  gp <- gpar(lwd = len0_null(size * .pt), col = colour, lty = linetype, lineend = lineend)
+  gp <- gpar(lwd = len0_null(size * .pt), col = colour, lty = linetype,
+             lineend = lineend, fill = colour)
```
Comment (clauswilke) [inline R/theme-elements.r]: Could you use tidyverse-style formatting? The same applies for the other change.

``` r
gp <- gpar(
  lwd = len0_null(size * .pt), col = colour, lty = linetype,
  lineend = lineend, fill = colour
)

Comment (clauswilke) [reply R/theme-elements.r]: And maybe reorder the parameters so `col` and `fill` go together and then `lwd`, `lty`, and `lineend`.

**Comment by clauswilke** [comment]
Thanks! This will have to wait until after the 3.1.0 release.

File: R/theme-elements.r (hunk 1 above)
Comment (yutannihilation) [reply R/theme-elements.r]: Sure.

**Comment by yutannihilation** [comment]
Thanks for reviewing! I'll wait (and will add a NEWS bullet after the release).

**Comment by yutannihilation** [comment]
Thanks!

**Pull request merged**